### PR TITLE
Depthwise_conv2d  optimization

### PR DESCRIPTION
**Pull request by zhangboSJTU**
### PR types

Performance optimization
### PR changes

OPs
### Describe

use cudnn for depthwise_conv2d

### CODE REVIEW

**Comment by zhangbopd** [comment]
> ❌ The PR is not created using [PR's template](https://github.com/PaddlePaddle/Paddle/blob/develop/.github/PULL_REQUEST_TEMPLATE.md). You can refer to this [Demo](https://github.com/PaddlePaddle/Paddle/pull/24877). Please use PR's template, it helps save our maintainers' time so that more developers get helped.

File: paddle/phi/kernels/gpudnn/conv_grad_kernel.cu [hunk 1]
```diff
@@ -770,10 +770,6 @@ void DepthwiseConvCudnnGradKernel(const Context& dev_ctx,
                                   int groups,
                                   const std::vector<int>& dilations,
                                   const std::string& data_format,
-                                  bool use_addto,
-                                  int workspace_size_MB,
-                                  bool exhaustive_search,
-                                  bool fuse_relu,
```
Comment (JamesLim-sy) [inline paddle/phi/kernels/gpudnn/conv_grad_kernel.cu:776]: 如果调用cuDNN 执行计算的话，这三项参数use_addto, workspace_size_MB, exhaustive_search, 都是cuDNN 自带的参数，不清楚为何删除

**Comment by JamesLim-sy** [comment]
可以仿照`PHI_DECLARE_bool(cudnn_deterministic);`做专用的优化，毕竟针对的是业务模型，框架那边离着业务模型远，意见没有那么重要.

**Comment by zhangboSJTU** [comment]
因为单测精度问题导致无法合入框架，手写 kernel 和调库精度存在diff

**Pull request closed without merging**